src/pybind/mgr/dashboard/frontend/src/app/shared/api/smb.service.ts

```diff
@@ -19,13 +26,52 @@ export class SmbService {
   baseURL = 'api/smb';
   private modalDataSubject = new Subject<DomainSettings>();
   modalData$ = this.modalDataSubject.asObservable();
+  dataUploader = signal<SMBResource | null>(null);
+  dataUploaderHider = signal<boolean>(false);
 
-  constructor(private http: HttpClient) {}
+  constructor(private http: HttpClient, private notificationService: NotificationService) {}
 
   passData(data: DomainSettings) {
     this.modalDataSubject.next(data);
   }
 
+  setDataUploaded(data: SMBResource): void {
+    this.dataUploader.set(data);
+  }
+
+  uploadData(event: any) {
+    const file = event.values().next().value?.file;
+    if (!file) {
+      return;
+    }
+
+    if (!(file.type === APPYAML || file.type === APPJSON)) {
+      this.notificationService.show(
+        NotificationType.error,
+        $localize`Invalid file type: only .json or .yaml accepted`
+      );
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.onload = (e: ProgressEvent<FileReader>) => {
+      const fileContents: any = e.target?.result;
+      let parsedContent: any;
+      try {
+        if (file.type === APPYAML) {
+          parsedContent = yaml.load(fileContents);
+        } else if (file.type === APPJSON) {
+          parsedContent = JSON.parse(fileContents);
+        }
+      } catch (error) {
+        this.notificationService.show(NotificationType.error, $localize`${error.message}`);
```

Comment:
same here, when do we show these messages?

Reply:
Whenever the yaml/json parsers raise an exception, meaning something went wrong while getting the information from the file.

In this example my yaml had a field wrongly tabbed, so I get the notification error message:

[wrong_formatted_data.webm](https://github.com/user-attachments/assets/37ab675e-4046-43b2-aaee-3cd3d2028386)